test/stdlib/Builtins.swift

```diff
@@ -293,4 +293,17 @@ tests.test("_isOptional") {
   expectFalse(_isOptional(P.self))
 }
 
+tests.test("_isConcrete") {
+  @_transparent
+  func isConcrete_true<T>(_ type: T.Type) -> Bool {
+    return _isConcrete(type)
+  }
+  func isConcrete_false<T>(_ type: T.Type) -> Bool {
```

Comment:
What would the effect of marking this `@inline(__always)` be? Does `isConcrete_true` require `@_transparent` specifically? If not, is this fragile to inlining decisions? (and so does `isConcrete_false` need to be marked `@inline(never)`?)

Reply:
`@inline(__always)` is *still* not fully implemented--it does nothing at `-Onone`. So `isConcrete_true` requires `@_transparent`. `isConcrete_false` is fragile to inlining decisions. It happens to work here because the test is only run at `-Onone`, but this assumption could be called out explicitly by adding `@inline(never)`.

Reply:
Let's please add the `@inline(never)` here, then.

Reply:
Why don't we test builtins in `-O` on top of `-Onone`?

Reply:
Any optimizations that might affect the builtin should definitely have their own tests. You've already done that for constant folding. Now that you've added the SILCombine optimization you should have corresponding test somewhere like `test/SILOptimizer/sil_combine.sil`

Reply:
Thanks! I think we just need Andy's suggested test, now.

Reply:
Added test to bottom of `test/SILOptimizer/sil_combine.sil`.